### PR TITLE
Add ipadm_ifprop module

### PR DESCRIPTION
**Pull request by xen0l**
##### ISSUE TYPE

 - New Module Pull Request

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY
Module for managing IP interface properties on illumos/Solaris systems.

### CODE REVIEW

File: lib/ansible/modules/network/illumos/ipadm_ifprop.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+ipadm_ifprop: protocol=ipv4 interface=e1000g0  temporary=true property=forwarding state=reset
+
+# Configure IPv6 metric on network interface e1000g0
+ipadm_ifprop: protocol=ipv6 nic=e1000g0 name=metric value=100
+
+# Set IPv6 MTU on network interface bge0
+ipadm_ifprop: interface=bge0 name=mtu value=1280 protocol=ipv6
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_ifprop.py]: return data must be documented

File: lib/ansible/modules/network/illumos/ipadm_ifprop.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+    if out:
+        result['stdout'] = out
+    if err:
+        result['stderr'] = err
+
+    module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_ifprop.py]: please use explicit imports .... `from ansible.module_utils.basic import AnsibleModule` ... etc

File: lib/ansible/modules/network/illumos/ipadm_ifprop.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+main()
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_ifprop.py]: please wrap in block

```
if __name__ == '__main__':
   main()
```

**Review by privateip — CHANGES_REQUESTED** [review]
* Missing metadata.  Metadata is now required for 2.3 modules
* please consider adding unit tests to the test folder as well.  this will be a required element soon

**Comment by xen0l** [comment]
ready_for_review

**Comment by xen0l** [comment]
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

**Review by privateip — APPROVED** [review]

**Pull request merged**